Pre clone scripts update (#2781)
codemagic-ci-cd/codemagic-docs: content/yaml-basic-configuration/yaml-getting-started.md

There are example scripts available for building a [Flutter application](./building-a-flutter-app/), [React Native application](./building-a-react-native-app/), [native Android application](./building-a-native-android-app/) or a [native iOS application](./building-a-native-ios-app/).

> To run scripts before the repository cloning step, the following syntax needs to be applied:

Reviewer: ```suggestion
By default, scripts are run after your app sources are fetched and the repository is cloned. To run scripts before the repository cloning step, use the pre_clone_scripts step:
```

{{< highlight yaml "style=paraiso-dark">}}
workflows:
    default-workflow:
        name: Default Workflow
        pre_clone_scripts:

>           - name: Scripts before repository cloned

Reviewer: ```suggestion
          - name: Scripts to run before cloning the repository
```

            script:
                 .......
        scripts:

>             - name: Scripts after repository cloned

Reviewer: ```suggestion
            - name: Scripts to run after cloning the repository
```

              script: |

>                  .......

Reviewer 2: I'm a little late to the party, but there are couple of styling issues here:
- There is mixed usage of 2 and 4 space indentation.
- `.......` as a placeholder value looks uncommon and weird in comparison to standard `...`.

Also in pre-clone scripts section the multiline script value is missing a pipe character, which renders this to be an invalid Yaml.

Altogether this would have been better:
```yaml
workflows:
  default-workflow:
    name: Default Workflow
    pre_clone_scripts:
      - name: Scripts to run before cloning the repository
        script: ...
    scripts:
      - name: Scripts to run after cloning the repository
        script: ...
```
